### PR TITLE
Add indentation rules

### PR DESCRIPTION
**Pull request by as0501**

### CODE REVIEW

File: src/Front/Scanner.x [hunk 1]
```diff
@@ -101,6 +101,7 @@ tokens :-
 
     -- If we're about to start a block, newlines don't matter.
     @newline            ;
+    -- @newline            { \_ _ -> alexSetStartCode blockLineStart >> alexMonadScan }
```
Comment (j-hui) [inline src/Front/Scanner.x:104]: Is this meant to be commented out?

File: src/Front/Scanner.x [hunk 2]
```diff
@@ -235,6 +251,7 @@ data AlexUserState = AlexUserState
   { usContext :: [ScannerContext] -- ^ stack of contexts
   , commentLevel :: Word          -- ^ 0 means no block comment
   , lastCtxCode :: Int            -- ^ last seen scanning code before block
+  , lastLineMargin :: Int
```
Comment (j-hui) [inline src/Front/Scanner.x:254]: Can you add documentation to this?

Comment (j-hui) [reply src/Front/Scanner.x:254]: Use the same `-- ^ documentation` Haddock comment format as the above fields

File: src/Front/Scanner.x [hunk 3]
```diff
@@ -348,6 +366,16 @@ gotoStartLine _ _ = do
   alexSetStartCode lineStart
   alexMonadScan
 
+extractMargin :: AlexUserState -> Int
```
Comment (j-hui) [inline src/Front/Scanner.x:369]: Can you document these functions? Use the `-- | documentation` Haddock comments as the other functions

File: src/Front/Scanner.x [hunk 4]
```diff
@@ -348,6 +366,16 @@ gotoStartLine _ _ = do
...
+extractMargin (AlexUserState _ _ _ m) = m
+
+updateMargin :: AlexInput -> Alex ()
```
Comment (j-hui) [inline src/Front/Scanner.x:372]: Document this

Comment (as0501) [reply src/Front/Scanner.x:373]: Thank you for the comments, just added documentation and cleaned it up!

**Pull request merged**